src/shared/Microsoft.Git.CredentialManager/Interop/LibGit2.cs

```diff
@@ -129,6 +173,7 @@ private void Dispose(bool disposing)
         {
             unsafe

```

Comment:
should we check the disposing flag?

Reply:
You're correct. In fact I've noticed a few other `IDisposable` objects not implementing the disposable pattern 100% correctly.

I'm going to create an abstract `DisposableObject` class that will do this and ensure all object in GCM Core that are disposable derive from this.